### PR TITLE
AmbientLightSensor: add some basic functionality tests

### PR DESCRIPTION
**Pull request by Honry**
Cover start(), stop(), onchange, onstatechange and also secure stuff when browsing context

### CODE REVIEW

File: ambient-light/AmbientLightSensor_start.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+<script>
+var sensor, start_return;
+
+setup(function() {
+  sensor = new AmbientLightSensor();
+  start_return = sensor.start();
+});
+
+test(function() {
+  assert_equals(sensor.state, "activating");
```
Comment (zqzhang) [inline ambient-light/AmbientLightSensor_start.html]: I think it is better to put all these assertions in this test file together within `ambient-light/AmbientLightSensor_onstatechange.html`.

File: ambient-light/AmbientLightSensor_stop.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+var sensor, stop_return;
+setup(function() {
+  sensor = new AmbientLightSensor();
+  sensor.start();
+  stop_return = sensor.stop();
+});
+
+test(function() {
+  assert_equals(sensor.state, "idle");
```
Comment (zqzhang) [inline ambient-light/AmbientLightSensor_stop.html]: I think it is better to put all these assertions in this test file together within `ambient-light/AmbientLightSensor_onstatechange.html`.

File: ambient-light/AmbientLightSensor_start.html (hunk 1 above)
Comment (Honry) [inline ambient-light/AmbientLightSensor_start.html]: @zqzhang, thanks for the comments, do you mean sub-case is unnecessary here?

Comment (Honry) [inline ambient-light/AmbientLightSensor_start.html]: @zqzhang, thanks for your comments! Agree!

**Comment by zqzhang** [comment]
@Honry I mean it is better to combine all the tests in `AmbientLightSensor_{onchange|start|stop}.html` into one test file, and to check things in each state, e.g. before and after start, before and after stop. Then the duplicates go away by nature.

**Comment by Honry** [comment]
@zqzhang, initially I designed 3 test files for onstatechange, start, and stop since they may be extended with more checkpoints in future, such as permission, 'active' state, reporting mode and etc., to make each test file distinct. WDYT?

**Pull request merged**